iXBRLViewerPlugin/iXBRLViewer.py

```diff
@@ -195,6 +198,24 @@ def getRelationships(self):
                 rels.setdefault(self.roleMap.getPrefix(arcrole),{})[self.roleMap.getPrefix(ELR)] = rr
         return rels
 
+    def validationErrors(self):
+        dts = self.dts
+
+        logHandler = dts.modelManager.cntlr.logHandler
+        if not hasattr(logHandler, "logRecordBuffer"):
+            raise IXBRLViewerBuilderError("Logging is not configured to use a buffer.  Unable to retrieve validation messages")
```

Comment:
We have internal use cases where we do not want to include validation issues in viewers.  It looks like we could override this method to just return empty, but is it better to exclude the key from the JSON altogether and/or to return None here?

@derekgengenbacher-wf FYI, we'll presumably need to adapt in some way when building the internal viewer.  I'm not sure if it's already using `logToBuffer`, but if so we'll need to be sure to somehow disable this.

Reply:
Should we make this an option on `createViewer` so that you can choose whether to include validation messages when creating the viewer?

I'm not sure how you're generating viewers at the moment, but as @derekgengenbacher-wf says, you may not be using `logToBuffer` anyway. 

It was surprisingly difficult to get hold of the messages from a validation run – I couldn't find anyway other than via this log-to-buffer dance.

Reply:
If we add that option we can default it to on so that most impls of this dont have to do anything, we can turn it off manually in our impls.

Reply:
This is now done in b91837c